test/unit/krane/resource_deployer_test.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+require 'test_helper'
+require 'krane/resource_deployer'
+
+class ResourceDeployerTest < Krane::TestCase
+  def test_deploy_prune_builds_whitelist
+    whitelist_kind = "fake_kind"
+    resource = build_mock_resource
+    Krane::Kubectl.any_instance.expects(:run).with do |*args|
+      args.include?("--prune-whitelist=#{whitelist_kind}")
+    end.returns(["", "", stub(success?: true)])
+    resource_deployer(kubectl_times: 0, prune_whitelist: [whitelist_kind]).deploy!([resource], false, true)
+  end
+
+  def test_deploy_no_prune_doesnt_prune
```

Comment:
I would have expected to see almost everything in this file as an integration test, not a unit test: this is the core functionality of the task. Is there a particular reason you chose unit testing this time?

Reply:
If I had written it from scratch I definitely would have made them integration tests. But between the existing integration tests exercising this code and having unit tests for many of the other classes I went this direction. Happy to make them integration tests.

Reply:
Please do make them integration tests.

> having unit tests for many of the other classes

Not for the tasks themselves though, right? In general I personally tried to use integration tests for everything except edge cases other behavior that can't easily be tested by an integration test.

Reply:
>Not for the tasks themselves though, right? 

I'm not sure I follow, this isn't a task and now that I look only the tasks have integration tests.

Reply:
Ah, you're right, sorry--not sure what made me think these were task tests. Does the GlobalDeployTask have integration coverage of these things? I only remember seeing one integration test for it, but maybe I missed something.

Reply:
Added additional integration tests to `test/integration/global_deploy_test.rb `  (I'll add ones that test pruning in the PR that adds pruning support.)

Reply:
Is-there/do-we-need a test to ensure a namespaced CR is _not_ deployed whereas a cluster-scope CR _is_?